File: audit/src/main/java/com/tek/audit/repository/WebAuditRepository.java
package com.tek.audit.repository;

import com.tek.audit.model.WebAudit;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

public interface WebAuditRepository extends JpaRepository<WebAudit, Long>,
    JpaSpecificationExecutor<WebAudit> {

  @Override
  Page<WebAudit> findAll(Specification<WebAudit> spec, Pageable pageable);
}

File: audit/src/main/kotlin/com/tek/audit/repository/WebAuditRepository.kt
package com.tek.audit.repository

import com.tek.audit.model.WebAudit
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.stereotype.Repository

@Repository
interface WebAuditRepository : JpaRepository<WebAudit, Long>, QuerydslPredicateExecutor<WebAudit>